src/core/public/application/ui/app_container.tsx

```diff
@@ -53,25 +53,27 @@ export const AppContainer: FunctionComponent<Props> = ({
         unmountRef.current = null;
       }
     };
-    const mount = async () => {
-      if (!mounter || appStatus !== AppStatus.accessible) {
-        return setAppNotFound(true);
-      }
 
-      if (mounter.unmountBeforeMounting) {
-        unmount();
-      }
+    if (!mounter || appStatus !== AppStatus.accessible) {
+      return setAppNotFound(true);
+    }
+    setAppNotFound(false);
 
+    if (mounter.unmountBeforeMounting) {
+      unmount();
+    }
```

Comment:
The main fix was to `setAppNotFound(false);` before awaiting for `mounter.mount`.
I also extracted all the non-mount related code outside of `mount` function to execute all we can in a synchronous way (effects does not support async, so we are not effectively waiting for `mount` to resolve)

Reply:
This is starting to feel like a class component would be simpler to understand.

Reply:
I agree. As discussed on slack, I think we should migrate back to class component next time we need to perform changing on this component. This may be when we'll want to show a proper loading message while awaiting the `mounter.mount` result.